Add an Interface for Configuration

solvingproblemswithtechnology opened the pull request:
* I added a WinForms Core project so we have an interface to change OBS settings.
* Started to generalizing OBS and PowerPoint interaction so maybe we can add support for google docs and others.
* Added a GitHub Action to generate a release on v* tags.
  * I had to remove the COMReference and use a Good Old Nuget™. I had many problems with this.

solvingproblemswithtechnology committed:
Creating project and the Form

solvingproblemswithtechnology committed:
Creating project and the Form

solvingproblemswithtechnology committed:
Initial ConfigurationFormLayour

solvingproblemswithtechnology committed:
Initial ConfigurationFormLayour

solvingproblemswithtechnology committed:
Interface changes

solvingproblemswithtechnology committed:
Interface changes

solvingproblemswithtechnology committed:
Ignore new project folder

solvingproblemswithtechnology committed:
Ignore new project folder

solvingproblemswithtechnology committed:
OBS Sspport

solvingproblemswithtechnology committed:
OBS Sspport

solvingproblemswithtechnology committed:
Powerpoint support

solvingproblemswithtechnology committed:
Powerpoint support

solvingproblemswithtechnology committed:
Move project to its parent folder (bad creation)

solvingproblemswithtechnology committed:
Move project to its parent folder (bad creation)

solvingproblemswithtechnology committed:
Inital changes for saving configuration and load at startup.

solvingproblemswithtechnology committed:
Inital changes for saving configuration and load at startup.

skipishere committed:
Fix exception handling code

solvingproblemswithtechnology committed:
Remove dependencies from the Form.
Handle the lifecycle of the client and settings in Program.

solvingproblemswithtechnology committed:
Remove dependencies from the Form.
Handle the lifecycle of the client and settings in Program.

solvingproblemswithtechnology committed:
Github Action

solvingproblemswithtechnology committed:
Github Action

solvingproblemswithtechnology committed:
Github Action

solvingproblemswithtechnology committed:
Rename

solvingproblemswithtechnology committed:
Rename error

solvingproblemswithtechnology committed:
Rename error

solvingproblemswithtechnology committed:
Update release-action.yml

Use msbuild instead of dotnet core due to COMReference

solvingproblemswithtechnology committed:
Update release-action.yml

Use msbuild instead of dotnet core due to COMReference

solvingproblemswithtechnology committed:
Update release-action.yml

Use another MSBuild.

solvingproblemswithtechnology committed:
Update release-action.yml

Use another MSBuild.

solvingproblemswithtechnology committed:
Github Action

solvingproblemswithtechnology committed:
Github Action

solvingproblemswithtechnology committed:
Using good old nuget instead of COM for CI...

solvingproblemswithtechnology committed:
Using good old nuget instead of COM for CI...

solvingproblemswithtechnology committed:
Using good old nuget instead of COM for CI...

solvingproblemswithtechnology committed:
Using good old nuget instead of COM for CI...

solvingproblemswithtechnology committed:
Merge branch 'main' of https://github.com/smartcodinghub/PowerPointToOBSSceneSwitcher into main

bparks committed:
Add special commands to start and stop recording; allow multiple commands per slide

bparks committed:
Avoid crashing if recording is already started/stopped and start/stop is requested

bparks committed:
Fix comment

shanselman committed:
Merge pull request #5 from skipishere/main

Fix exception handling code

shanselman commented:
This looks amazing! Given mine is a little console app and this is really a re-write or complete refactoring, how about this? 

Take your Fork and rename something like PowerPointToOBS-WinForms or something and I'll LINK to you from my readme? That way there's two - console and forms?

solvingproblemswithtechnology commented:
Thank you!

Given that the idea is yours and you also did a video (which I enjoyed) I think it's fine to contribute a little. I had a similar WinForms app (also with an action) so It was fast to add it to the project. Also, it will be way more useful in your repo.

I like the idea of having both, a Console and a Form. Maybe can we use https://github.com/natemcmaster/CommandLineUtils or something like that? And create a single executable?

Anyway, go ahead and merge it if you like, that's the spirit of open source :)

P.D.: Sorry for the english :)

shanselman committed:
Merge pull request #8 from bparks/control-start-stop

Add special commands to start and stop recording

shanselman commented:
I don't know how to make an app that is both WinForms and Console. Maybe a second project in this solution? 
Perdon por mi español ;)

solvingproblemswithtechnology commented:
Hi!

Merge the PR (if you like) and I'll do another one rebasing the new code and with the Console+WinForms project.

solvingproblemswithtechnology committed:
Refactor to Console+Winforms app.

solvingproblemswithtechnology committed:
Start/Stop feature.

solvingproblemswithtechnology committed:
Add new project for generating an dll embed with the PowerPoint COMReference

solvingproblemswithtechnology committed:
Rebasing

solvingproblemswithtechnology committed:
Creating a console project for optional GUI.

solvingproblemswithtechnology committed:
Fix spelling for action

solvingproblemswithtechnology committed:
Fix dotnet warp output name.

solvingproblemswithtechnology committed:
dotnet warp can't create a path...

solvingproblemswithtechnology commented:
I updated with:

* Brand new console app using https://github.com/natemcmaster/CommandLineUtils
  * Can use the --help option to see all the options.
  * Can use --no-gui to run the app without gui.
  * Added DI and standard logging.
* Converting the winforms project to dll
* Added dotnet warp to generate the release in github action.
* Added the start/finish OBS command.
* ==> Created a project so I can have the COM Reference embed in the dll. This allows using the dotnet cli instead of msbuild.
  * The nuget worked until I cleaned the solution, so I had to do this thing. Anyway, is a good way for managing interops.

@shanselman My only concern now is about naming these projects :) Naming is hard.

solvingproblemswithtechnology committed:
Exclude project directory.
